nexus/db-macros/src/lookup.rs

```diff
@@ -287,63 +284,70 @@ fn generate_struct(config: &Config) -> TokenStream {
     }
 }
 
-/// Generates the child selectors for this resource
+/// Generates the child selector for this resource's parent
 ///
-/// For example, for the "Project" resource with child resources "Instance" and
-/// "Disk", this will generate the `Project::instance_name()` and
-/// `Project::disk_name()` functions.
-fn generate_child_selectors(config: &Config) -> TokenStream {
-    let child_resource_types: Vec<_> =
-        config.child_resources.iter().map(|c| format_ident!("{}", c)).collect();
-    let child_selector_fn_names: Vec<_> = config
-        .child_resources
-        .iter()
-        .map(|c| format_ident!("{}_name", heck::AsSnakeCase(c).to_string()))
-        .collect();
-    let child_selector_fn_names_owned: Vec<_> = config
-        .child_resources
-        .iter()
-        .map(|c| {
-            format_ident!("{}_name_owned", heck::AsSnakeCase(c).to_string())
-        })
-        .collect();
-    let child_selector_fn_docs: Vec<_> = config
-        .child_resources
-        .iter()
-        .map(|child| {
-            format!(
-                "Select a resource of type {} within this {}, \
-                identified by its name",
-                child, config.resource.name,
-            )
-        })
-        .collect();
+/// For example, for the "Instance" resource with parent resource "Project",
+/// this will generate the `Project::instance_name()` and
+/// `Project::instance_name_owned()` functions.
+///
+/// This is generated by the child resource codegen, rather than by the parent
+/// resource codegen, since such functions are only generated for resources with
+/// `lookup_by_name = true`.  Whether or not this is enabled for the child
+/// resource is not known when generating the parent resource code, so it's
+/// performed by the child instead.
+fn generate_child_selector(config: &Config) -> TokenStream {
+    // If this resource can only be looked up by ID, we don't need to generate
```

Comment:
Suuuuuuper nitpicky, but: 
```suggestion
    // If this resource cannot be looked up by name, we don't need to generate
```

(There could plausibly be a non-UUID, non-name lookup)